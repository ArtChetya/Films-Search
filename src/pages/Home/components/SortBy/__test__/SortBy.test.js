import { shallow } from 'enzyme/build';
import React from 'react';
import { SortBy } from '..';

describe('SortBy component', () => {
    let options;
    let selectedOptionId;
    let onOptionChange;

    beforeAll(() => {
        options = [
            { id: 'title', label: 'Title' },
            { id: 'genres', label: 'Genre' }
        ];

        selectedOptionId = 'title';

        onOptionChange = jest.fn();
    });

    it('should be rendered correctly', () => {
        const component = shallow(
            <SortBy
                options={options}
                selectedOptionId={selectedOptionId}
                onOptionChange={onOptionChange}
            />
        );
        expect(component).toMatchSnapshot();
    });
});
