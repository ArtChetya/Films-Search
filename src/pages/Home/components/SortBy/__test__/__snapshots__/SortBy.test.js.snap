// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy component should be rendered correctly 1`] = `
<FilterBy
  config={
    Object {
      "activeBgColor": "transparent",
      "activeColor": "#ec407a",
      "buttonTextTransform": "lowercase",
      "buttonVariant": "text",
      "defaultBgColor": "transparent",
      "defaultColor": "#616161",
      "title": "Sort by",
      "titleTextTransform": "none",
      "titleVariant": "subtitle2",
    }
  }
  onOptionChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "title",
        "label": "Title",
      },
      Object {
        "id": "genres",
        "label": "Genre",
      },
    ]
  }
  selectedOptionId="title"
/>
`;
